### PR TITLE
refactor(neon): Make account selection reusable

### PR DESCRIPTION
**Pull request by provokateurin**
Fixes https://github.com/nextcloud/neon/issues/598

Split off from https://github.com/nextcloud/neon/pull/600

### CODE REVIEW

File: packages/neon/neon/lib/src/widgets/account_selection_dialog.dart [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+  @override
+  Widget build(final BuildContext context) {
+    final accountsBloc = Provider.of<AccountsBloc>(context, listen: false);
+    final accounts = accountsBloc.accounts.value;
+    final activeAccount = accountsBloc.activeAccount.value!;
+
+    final inactiveAccounts = List.of(accounts)..removeWhere((final account) => account.id == activeAccount.id);
+
+    final tiles = [activeAccount, ...inactiveAccounts].map(
```
Comment (provokateurin) [inline packages/neon/neon/lib/src/widgets/account_selection_dialog.dart:27]: Is there a better way to do this? The active account just needs to be the first in the list. Would sorting make more sense?

Comment (Leptopoda) [reply packages/neon/neon/lib/src/widgets/account_selection_dialog.dart:27]: I think:
```dart
    final sortedAccounts = List.of(accounts)
      ..removeWhere((final account) => account.id == activeAccount.id)
      ..insert(0, activeAccount);
```

Makes most sense here.

File: packages/neon/neon/lib/src/widgets/account_selection_dialog.dart [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+            Navigator.of(context).pop(account);
+          },
+        );
+        if (highlight) {
+          widget = Column(
+            children: [
+              widget,
+              const Divider(),
+            ],
+          );
```
Comment (Leptopoda) [inline packages/neon/neon/lib/src/widgets/account_selection_dialog.dart:43]: I don't like nested columns.

Maybe jsut call `tiles.insert(1, const Divider())` after the map.

File: packages/neon/neon/lib/src/widgets/account_selection_dialog.dart [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+          (final account) => NeonAccountTile(
+            account: account,
+            trailing: highlightActiveAccount && account.id == activeAccount.id ? const Icon(Icons.check_circle) : null,
+            onTap: () {
+              Navigator.of(context).pop(account);
+            },
+          ),
+        )
+        .toList();
+    if (highlightActiveAccount) {
```
Comment (Leptopoda) [inline packages/neon/neon/lib/src/widgets/account_selection_dialog.dart:40]:
```suggestion
    if (highlightActiveAccount && accounts.length > 1) {
```

**Review by Leptopoda — APPROVED** [review]

**Pull request merged**